src/sota_tools/CMakeLists.txt

```diff
@@ -11,19 +10,100 @@ set (GARAGE_PUSH_SRCS
   ostree_object.cc
   request_pool.cc
   treehub_server.cc
+  deploy.cc
+)
+
+add_library(sota_tools_static_lib STATIC ${SOTA_TOOLS_LIB_SRC})
+
+##### garage-push targets
+
+set (GARAGE_PUSH_SRCS
+  garage_push.cc
 )
 
+add_executable(garage-push ${GARAGE_PUSH_SRCS})
 
-add_executable(garage-push main.cc)
-add_library(garage_push_static_lib STATIC ${GARAGE_PUSH_SRCS})
-target_link_libraries(garage-push garage_push_static_lib aktualizr_static_lib
+# Export compile_commands.json for clang-check
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+target_compile_options(sota_tools_static_lib PUBLIC -std=c++11)
+
+target_include_directories(sota_tools_static_lib INTERFACE ${PROJECT_SOURCE_DIR} ${GLIB2_INCLUDE_DIRS})
```

Comment:
I had to change this line for a fix in meta-updater. See my https://github.com/advancedtelematic/aktualizr/pull/384/commits/a0c50801d1a9472ccbf7c8d73d9032529913ac31 for the details. You should keep the syntax from that but of course use the new `sota_tools_static_lib` name.